### PR TITLE
Functional test: Check out items

### PR DESCRIPTION
**Pull request by thofmann**
### Summary

Functional tests for the checkout API.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Check the tests.
2. Test the tests.

### Links

- [TFF-134](https://msoese.atlassian.net/browse/TFF-134)

### CODE REVIEW

File: src/api/checkout.js [hunk 1]
```diff
@@ -4,6 +4,13 @@ import { addAction } from '../lib/database';
 let app = express();
 
 app.post('/', (req, res) => {
+    if (typeof req.body.studentId !== 'string') {
+        return res.failureJson('A student id is required.');
```
Comment (mattmahn) [inline src/api/checkout.js]: `res.status(400).failureJson`

File: src/api/checkout.js [hunk 2]
```diff
@@ -4,6 +4,13 @@ import { addAction } from '../lib/database';
...
+    }
+    if (typeof req.body.itemAddresses !== 'object'
+        || typeof req.body.itemAddresses[0] !== 'string') {
+        return res.failureJson('An array of item addresses is required.');
```
Comment (mattmahn) [inline src/api/checkout.js]: `res.status(400).failureJson`

File: test/functional/checkout.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        }).then(data => {
+            assert.isUndefined(data);
+            assert.lengthOf(CheckoutStore.getCheckouts(), 1);
+            assert.lengthOf(ItemStore.getItems().filter(item => item.status === 'CHECKED_OUT'), 1);
+            assert.lengthOf(ItemStore.getItems().filter(item => item.status === 'AVAILABLE'), 0);
+            assert.strictEqual(StudentStore.getStudentById('123456').items[0].address, 'iGwEZUvfA');
+        });
+    });
+
+    it('should mot be able to check out an unavailable item', () => {
```
Comment (tondef001) [inline test/functional/checkout.js]: This typo should mot be allowed through

Comment (thofmann) [inline test/functional/checkout.js]: That's a nit.

File: src/api/checkout.js (hunk 1 above)
Comment (thofmann) [inline src/api/checkout.js]: Or not. This is completely optional, right?

Comment (mattmahn) [inline src/api/checkout.js]: Optional in the sense that nobody is forcing it. Required in the sense that it doesn't conform to the HTTP spec.

**Pull request merged**